04/Components/App.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import { v4 as uuid } from "uuid";
+
+import File from "./File";
+import List from "./List";
+
+
+export default class App extends React.Component {
+    fileRef = React.createRef();
+
+    state = {
+        filesList: [],
+    }
+
+    handleFileLoad = () => {
+        const files = this.fileRef.current.getFiles();
+
+        for (const file of files) {
+            if (file && file.type.includes('text/plain')) {
+                this.readFile(file);
+            } else {
+                alert(`${file.name} has wrong file type!`);
+            }
+        }
+    }
+
+    readFile(file) {
+        let reader = new FileReader();
+
+        reader.onload = () => {
+            const { name: fileName, size: fileSize } = file;
+            const fileContent = reader.result;
+            const fileObject = { fileName, fileSize, fileContent };
+
+            this.setFilesListState(fileObject);
+        }
+
+        reader.readAsText(file);
+    }
+
+    setFilesListState(fileData) {
+        this.setState(state => {
+            return {
+                filesList: [...state.filesList, fileData],
+            };
+        });
+    }
+
+    render() {
+        const { filesList } = this.state;
+        const filesItems = filesList.map(
+            ({ fileName, fileSize, fileContent }) =>
+                <li key={uuid()}>
```

Comment:
Podobnie jak wcześniej `uuid()` dodajmy na etapie zapisu do `state` i tutaj wykorzystujemy tj. `<li key={id}>`

Reply:
Dodałem na etapie odczytu pliku. Czy to będzie ok, czy jednak łamie zasadę pojedynczej odpowiedzialności?

Reply:
Tak, będzie ok